lib/routes/security_pin/security_and_backup/backup_tiles/backup_provider_tile.dart

```diff
@@ -0,0 +1,151 @@
+  final Future Function(BackupSettings backupSettings) backupNow;
+  final Function(dynamic exception) onError;
+
+  const BackupProviderTile({
+    @required this.backupSettings,
+    @required this.autoSizeGroup,
+    @required this.enterRemoteServerCredentials,
+    @required this.backupNow,
+    @required this.onError,
+  });
+
+  @override
+  State<BackupProviderTile> createState() => _BackupProviderTileState();
+}
+
+class _BackupProviderTileState extends State<BackupProviderTile> {
+  @override
+  Widget build(BuildContext context) {
+    final backupBloc = AppBlocsProvider.of<BackupBloc>(context);
+
+    final texts = context.texts();
+    final currentProvider = widget.backupSettings.backupProvider;
+
+    return ListTile(
+      title: AutoSizeText(
+        texts.security_and_backup_store_location,
+        style: const TextStyle(color: Colors.white),
+        maxLines: 1,
+        minFontSize: MinFontSize(context).minFontSize,
+        stepGranularity: 0.1,
+        group: widget.autoSizeGroup,
+      ),
+      trailing: DropdownButtonHideUnderline(
+        child: DropdownButton<BackupProvider>(
+          iconEnabledColor: Colors.white,
+          value: currentProvider,
+          isDense: true,
+          onChanged: (BackupProvider selectedProvider) async {
+            if (selectedProvider.isGDrive) {
+              await _logoutWarningDialog(currentProvider).then(
+                (logoutApproved) async {
+                  if (logoutApproved) {
+                    try {
+                      EasyLoading.show();
+
+                      backupBloc.backupServiceNeedLoginSink.add(true);
+                      await _updateBackupProvider(selectedProvider);
+                    } catch (e) {
+                      log.warning("Failed to re-login & backup.", e);
+                      widget.onError(e);
+                    }
+                  }
+                },
+              );
+            }
+            if (currentProvider != selectedProvider) {
+              await _updateBackupProvider(selectedProvider);
+            }
+          },
+          items: BackupSettings.availableBackupProviders().map(
+            (provider) {
+              return DropdownMenuItem(
+                value: provider,
+                child: AutoSizeText(
+                  provider.displayName,
+                  style: theme.FieldTextStyle.textStyle,
+                  maxLines: 1,
+                  minFontSize: MinFontSize(context).minFontSize,
+                  stepGranularity: 0.1,
+                ),
+              );
+            },
+          ).toList(),
+        ),
+      ),
+    );
+  }
+
+  Future _logoutWarningDialog(BackupProvider previousProvider) async {
+    final backupBloc = AppBlocsProvider.of<BackupBloc>(context);
+    return await backupBloc.backupStateStream.first.then(
+      (backupState) async {
+        if ((backupState != BackupState.start() && previousProvider.isGDrive) ||
+            previousProvider.isGDrive) {
```

Comment:
The first condition should only be checked on Android platforms where default provider is set to Google Drive, otherwise the OR condition bypasses this check, prompting the user with this warning on a new account.

Also see EnableBackupDialog: https://github.com/breez/breezmobile/blob/7e1888598562cf6a6a3ac9954d923231123320a1/lib/widgets/enable_backup_dialog.dart#L263

cc: @ubbabeck

Reply:
Resolved, removed the obsolete OR.
This condition was written in a way to avoid platform specific imports/conditions: `backupState != BackupState.start() && previousProvider.isGDrive`
as this only applies to Android, the initial backup state on iOS will never have previousProvider as GDrive.